docs/shared/recipes/nx-release/update-local-registry-setup.md

```diff
@@ -0,0 +1,85 @@
+# Update Your Local Registry Setup to use Nx Release
+
+Nx will create a `tools/start-local-registry.ts` script for starting a local registry and publishing packages to it in preparation for running end to end tests. If you have an existing `tools/start-local-registry.ts` script from a previous version of Nx, you should update it to use Nx Release to publish packages to the local registry. This will ensure that newly generated libraries are published appropriately when running end to end tests.
+
+## The Previous Version
+
+The previous version of the `tools/start-local-registry.ts` script used publish targets on each project to publish the packages to the local registry. This is no longer necessary with Nx Release. You can identify the previous version by the `nx run-many` command that publishes the packages:
+
+```typescript
+/**
+ * This script starts a local registry for e2e testing purposes.
+ * It is meant to be called in jest's globalSetup.
+ */
+import { startLocalRegistry } from '@nx/js/plugins/jest/local-registry';
+import { execFileSync } from 'child_process';
+
+export default async () => {
+  // local registry target to run
+  const localRegistryTarget = '@demo-plugin-1800/source:local-registry';
+  // storage folder for the local registry
+  const storage = './tmp/local-registry/storage';
+
+  global.stopLocalRegistry = await startLocalRegistry({
+    localRegistryTarget,
+    storage,
+    verbose: false,
+  });
+  const nx = require.resolve('nx');
+  execFileSync(
+    nx,
+    ['run-many', '--targets', 'publish', '--ver', '0.0.0-e2e', '--tag', 'e2e'],
+    { env: process.env, stdio: 'inherit' }
+  );
+};
+```
+
+If your script looks like this, you should update it.
+
+## The Updated Version
+
+The updated version of the `tools/start-local-registry.ts` script uses Nx Release to publish the packages to the local registry. This is done by running `releaseVersion` and `releasePublish` functions from `nx/release`. Your updated script should look like this:
+
+```typescript
+/**
+ * This script starts a local registry for e2e testing purposes.
+ * It is meant to be called in jest's globalSetup.
+ */
+import { startLocalRegistry } from '@nx/js/plugins/jest/local-registry';
+import { execFileSync } from 'child_process';
+import { releasePublish, releaseVersion } from 'nx/release';
+
+export default async () => {
+  // local registry target to run
+  const localRegistryTarget = '@demo-plugin-1800/source:local-registry';
+  // storage folder for the local registry
+  const storage = './tmp/local-registry/storage';
+
+  global.stopLocalRegistry = await startLocalRegistry({
+    localRegistryTarget,
+    storage,
+    verbose: false,
+  });
+
+  const nx = require.resolve('nx');
+  execFileSync(nx, ['run-many', '--targets', 'build'], {
+    env: process.env,
+    stdio: 'inherit',
+  });
```

Comment:
Should this be removed in favor of the preversion command?

Reply:
I thought about this, but I wanted to make sure the e2e tests don't break if they edit or change the preVersionCommand in `nx.json` in the process of updating their _real_, non-e2e release process

Reply:
I figured in most cases, build will be cached, so the redundant run wouldn't hurt most of the time.

Reply:
I'm a bit mixed on this tbh. I get that line of logic, but also feel like the whole point of the verdaccio setup is to be **_as close to_** a regular release as possible. I think it makes sense that if they change the non-e2e release in a way that would break, the e2e should fail.

Reply:
ok. I see what you're saying. I'll go ahead and remove this line then.